tests/test_walker.py

```diff
@@ -0,0 +1,59 @@
+"""
+Test the walker infrastructure to read and execute recipes
+"""
+import os
+import glob
+import json
+import astropy.time as time
+import corgidrp
+import corgidrp.data as data
+import corgidrp.mocks as mocks
+import corgidrp.caldb as caldb
+import corgidrp.walker as walker
+
+
+def test_autoreducing():
+    """
+    Tests both generating and processing a basic L1->L2a SCI recipe
+    """
+    # create dirs
+    datadir = os.path.join(os.path.dirname(__file__), "simdata")
+    if not os.path.exists(datadir):
+        os.mkdir(datadir)
+    outputdir = os.path.join(os.path.dirname(__file__), "walker_output")
+    if not os.path.exists(outputdir):
+        os.mkdir(outputdir)
+
+
+    # create simulated data
+    l1_dataset = mocks.create_prescan_files(filedir=datadir, obstype="SCI", numfiles=2)
+    # fake the emgain
+    for image in l1_dataset:
+        image.ext_hdr['EMGAIN'] = 1
+    l1_dataset.save(datadir)
+    filelist = [frame.filepath for frame in l1_dataset]
+
+
+    # index the sample nonlinearity correction that we need for processing
+    new_nonlinearity = data.NonLinearityCalibration(os.path.join(os.path.dirname(__file__),"test_data",'nonlin_sample.fits'))
+    # fake the headers because this frame doesn't have the proper headers
+    prihdr, exthdr = mocks.create_default_headers("SCI")
+    new_nonlinearity.pri_hdr = prihdr
+    new_nonlinearity.ext_hdr = exthdr
+    new_nonlinearity.ext_hdr.set('DRPCTIME', time.Time.now().isot, "When this file was saved")
+    new_nonlinearity.ext_hdr.set('DRPVERSN', corgidrp.version, "corgidrp version that produced this file")
+    mycaldb = caldb.CalDB()
+    mycaldb.create_entry(new_nonlinearity)
+
+    # generate recipe
+    recipe = walker.autogen_reicpe(filelist, outputdir)
+
+    # process_recipe
+    walker.run_recipe(recipe)
+
```

Comment:
I think want a test or two here at the end to make sure that it has run correctly. At the least if we check that the two new files were saved that ensures the recipe was run to the end.

Reply:
Just added some checks to ensure the data can be loaded, and the recipe is written into the header!